### PR TITLE
Add DartPad best practices page

### LINKED ISSUE #2027: Add page of best practices for using DartPad in tutorials
Initially, this will be a PDF page produced from a Google Doc. I'll also need to add links to it from https://dart.dev/tools/dartpad, https://dart.dev/codelabs/async-await, and https://github.com/dart-lang/dart-pad/wiki.

/cc @galeyang

### PR DESCRIPTION
**Pull request by kwalrath**
...and link to it.
Also, make it easier to get to the main DartPad page from its troubleshooting page.

Fixes #2027

Staged:
https://kw-www-dartlang-1.firebaseapp.com/tools/dartpad
https://kw-www-dartlang-1.firebaseapp.com/codelabs/async-await#whats-next
https://kw-www-dartlang-1.firebaseapp.com/resources/dartpad-tutorials.pdf

### CODE REVIEW

**Comment by galeyang** [comment]
Thanks! I added two minor notes on your copy used for PDF conversion.

cc: @InMatrix

**Comment by kwalrath** [comment]
> Thanks! I added two minor notes on your copy used for PDF conversion.

updated, ptal

**Comment by kwalrath** [comment]
I'm going to modify the headers and add some copyright/license text, so let's not ship this quite yet.

**Comment by kwalrath** [comment]
fyi, I'm still waiting for word on the text we should use.

**Comment by kwalrath** [comment]
I plan to post this Tuesday.

**Comment by galeyang** [comment]
Just wanted to follow up on this :)

**Comment by kwalrath** [comment]
I still haven't heard from the person I'm supposed to hear from, and I've pinged them a couple of times. Time to give one last warning, then go ahead and post, I guess.

**Comment by kwalrath** [comment]
I've heard back... should get a definitive answer soon.

**Comment by chalin** [comment]
Btw, there's a typo in the PDF:

> Could Shell Walkthrough Writing Guide

should read

> Cloud Shell Walkthrough Writing Guide

File: src/tools/dartpad/index.md [hunk 1]
```diff
@@ -136,3 +136,22 @@ for (var char in 'hello'.split('')) {
...
+You can embed DartPad inside of web pages,
+customizing it to suit your purpose.
+For example, the [futures codelab][]
+contains multiple embedded DartPads
+labeled as _examples_ and _exercises_.
+
+For more information about how to use embedded DartPads, see
+[best practices for using DartPad in tutorials][].
+
+[futures codelab]: /codelabs/async-await
```
Comment (chalin) [inline src/tools/dartpad/index.md]: Nit: move this to the end of the page along with the other named link definitions?

**Pull request merged**